### PR TITLE
restore reconnect from observable flow

### PR DESCRIPTION
**Pull request by jimmcgaw**
Observables are such that I don't trust myself to figure out issues with them quickly. This restores the original functionality and hence the reconnect error, if we want an immediate fix.

### CODE REVIEW

File: src/store/epics/chatEpics.js [hunk 1]
```diff
@@ -76,31 +81,100 @@ const initChatHandlerConnection = ({ authToken, secretKey, mID, isNewRoom }) =>
...
+            chatHandler.getUserStatusSubject()
+              .map(setUserStatus),
+
+            Observable.of(connectionInitiated(), alertSuccess(`${isNewRoom ? 'New room created.' : ''} Connected to server.`, connectionAlertTtlSeconds))
+          )
+        ).catch(error => {
+          console.error(error);
+          return Observable.from([
+            alertWarning('Lost connection to chat server. Trying to reconnect...'),
+            disconnected()
```
Comment (jimmcgaw) [inline src/store/epics/chatEpics.js:174]: This sets the app to `shouldConnect`, which prompts a reconnection attempt, in part. This catch block is what initiates the retry, if the ajax request fails. This is what needs to be re-created, functionally, over in `sessions.js`.

**Comment by elimisteve** [comment]
In this branch I'm getting

```
ERROR in ./src/utils/sessions.js 7:0-54
Module not found: Error: Can't resolve '../actions/chatActions' in '/home/user/gocode/src/github.com/cryptag/leapchat/src/utils'
resolve '../actions/chatActions' in '/home/user/gocode/src/github.com/cryptag/leapchat/src/utils'
  using description file: /home/user/gocode/src/github.com/cryptag/leapchat/package.json (relative path: ./src/utils)
```

trying to build it.

**Comment by elimisteve** [comment]
Nevermind; that was a couple commits ago, I think.

Working now! Nice. Shall I merge this and close https://github.com/cryptag/leapchat/pull/322 without merging?

File: src/store/epics/chatEpics.js (hunk 1 above)
Comment (elimisteve) [reply src/store/epics/chatEpics.js:174]: Gotcha

**Review by elimisteve — APPROVED** [review]
LGTM; let's merge and I'll deploy :rocket:

**Comment by elimisteve** [comment]
_Gogogo_

**Pull request merged**